lib/bolt/executor.rb

```diff
@@ -212,6 +216,7 @@ def run_task(targets, task, arguments, options = {}, &callback)
       log_action(description, targets) do
         notify = proc { |event| @notifier.notify(callback, event) if callback }
         options = { '_run_as' => run_as }.merge(options) if run_as
+        options = { '_load_config' => @load_config }.merge(options) if @load_config
```

Comment:
https://github.com/puppetlabs/bolt/blob/master/lib/bolt/applicator.rb#L198 uses `batch_task` directly. If you make `_load_config` an opt-in check, then we also need to pass it there or `apply` won't use SSH config. Alternatively we could make it an opt-in by default (true unless passed false). I prefer opt-in by default, because I think it makes sense in most use-cases.

Reply:
In `with_connection` it does default to true. Will that get overwritten with 'nil' if it's passed?  I think my false assumption here was that if passed 'nil' it'd set the default, but does it actually get nil which is falsey?

Reply:
I think it gets nil.

Reply:
I might've been misreading the whole thing. I think what's here is correct, but probably has more checks than it needs.

Reply:
(shrug) Better safe than sorry.